### PR TITLE
fix(loggly): fix the error caused by the missing /bin/hostname

### PR DESCRIPTION
**Pull request by Water-Melon**
### Summary

1. Refactor the PDK's `get_hostname` to use only the `gethostname` system call to retrieve the hostname.
2. Loggly uses the PDK's `get_hostname` to retrieve the hostname.

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

FTI-6046

### CODE REVIEW

File: kong/plugins/loggly/handler.lua [hunk 1]
```diff
@@ -1,6 +1,7 @@
 local cjson = require "cjson"
 local sandbox = require "kong.tools.sandbox".sandbox
 local kong_meta = require "kong.meta"
+local get_host_name = require "kong.pdk.node".new().get_hostname
```
Comment (ms2008) [inline kong/plugins/loggly/handler.lua:4]:
```suggestion
local get_host_name = kong.node.get_hostname
```

**Review by ms2008 — COMMENTED** [review]
It is recommended that if the first call succeeds, we can cache the hostname in PDK so that there is no need to generate a system call for the next calls.

File: changelog/unreleased/kong/fix-loggly-hostname-notfound.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+message: "**loggly**: fixed the error caused by the missing `/bin/hostname`."
```
Comment (ms2008) [inline changelog/unreleased/kong/fix-loggly-hostname-notfound.yml:1]:
```suggestion
message: "**Loggly**: Fixed an issue where `/bin/hostname` missing caused an error warning on startup.
```

Comment (Water-Melon) [reply changelog/unreleased/kong/fix-loggly-hostname-notfound.yml]: updated.

**Review by ms2008 — APPROVED** [review]

**Review by oowl — APPROVED** [review]

**Pull request merged**